### PR TITLE
Semantical call info

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: crates/ra_ide/src/call_info.rs [hunk 1]
```diff
@@ -24,105 +46,130 @@ pub(crate) fn call_info(db: &RootDatabase, position: FilePosition) -> Option<Cal
...
-        FnCallNode::MethodCallExpr(method_call) => {
-            let function = sema.resolve_method_call(&method_call)?;
-            FunctionSignature::from_hir(sema.db, function)
-        }
-        FnCallNode::MacroCallExpr(macro_call) => {
-            let macro_def = sema.resolve_macro_call(&macro_call)?;
-            FunctionSignature::from_macro(sema.db, macro_def)?
+    let callable = match &calling_node {
+        FnCallNode::CallExpr(call) => sema.type_of_expr(&call.expr()?)?.as_callable(sema.db)?,
+        FnCallNode::MethodCallExpr(call) => sema.resolve_method_call_as_callable(call)?,
```
Comment (kjeremy) [inline crates/ra_ide/src/call_info.rs:116]: Did you remove macro call support on purpose?

Comment (matklad) [reply crates/ra_ide/src/call_info.rs:116]: Sort-of:

* I am not sure that signature help for macro is all that useful, given that we can’t even show parameters
* the implementation is cleaner without macros

So, I am neutral on having or not having it, and, during refactor, it was simpler to remove.

Comment (kjeremy) [reply crates/ra_ide/src/call_info.rs:116]: Makes sense!

**Pull request merged**